genie-web/src/main/java/com/netflix/genie/web/agent/launchers/impl/LocalAgentLauncherImpl.java

```diff
@@ -102,10 +117,43 @@ public LocalAgentLauncherImpl(
      */
     @Override
     public void launchAgent(@Valid final ResolvedJob resolvedJob) throws AgentLaunchException {
+        // Check error conditions
+        final int jobMemory = resolvedJob.getJobEnvironment().getMemory();
+        final String jobId = resolvedJob.getJobSpecification().getJob().getId();
+
+        // Job was resolved with more memory allocated than the system was configured to allow
+        if (jobMemory > this.launcherProperties.getMaxJobMemory()) {
+            throw new AgentLaunchException(
+                "Unable to launch job as the requested job memory ("
+                    + jobMemory
+                    + "MB) exceeds the maximum allowed by the configuration of the system ("
+                    + this.launcherProperties.getMaxJobMemory()
+                    + "MB)"
+            );
+        }
+
+        // One at a time to ensure we don't overflow configured max
+        synchronized (MEMORY_CHECK_LOCK) {
```

Comment:
Maybe leave a TODO or note here as a reminder that this does not guarantee enforcement of the limit.

Also after reading this, i think `getUsedMemoryOnHost` should include more than `RUNNING` as per comment on previous commit.